### PR TITLE
Improving test coverage for Vault & YearnStrategy contracts

### PR DESCRIPTION
**Pull request by brkomir**
Added several edge case tests to improve test code coverage of Vault and YearnStrategy contracts.

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -668,9 +668,6 @@ contract Vault is
         uint256 idsLen = _ids.length;
 
         for (uint256 i; i < idsLen; ++i) {
-            if (_amounts[i] > deposits[_ids[i]].amount)
-                revert VaultAmountTooLarge();
-
```
Comment (brkomir) [inline contracts/Vault.sol:673]: This is removed because this code was not tested by any test and the same check is present in the function _withdrawSingle on line 930 (033 on the main branch) 

- [if (_deposit.amount < _amount)
            revert VaultCannotWithdrawMoreThanAvailable(); ]

File: contracts/strategy/yearn/YearnStrategy.sol [hunk 2]
```diff
@@ -123,19 +123,13 @@ contract YearnStrategy is IStrategy, AccessControl, CustomErrors {
         onlyManager
     {
         if (amount == 0) revert StrategyAmountZero();
-        uint256 uninvestedUnderlying = _getUnderlyingBalance();
 
-        if (amount > uninvestedUnderlying) {
```
Comment (brkomir) [inline contracts/strategy/yearn/YearnStrategy.sol:128]: this logic is removed because all of the funds are immediately invested into yearn in `invest()` function

File: contracts/Vault.sol (hunk 1 above)
Comment (gabrielpoca) [reply contracts/Vault.sol:673]: well done 👍

File: contracts/strategy/yearn/YearnStrategy.sol (hunk 2 above)
Comment (gabrielpoca) [reply contracts/strategy/yearn/YearnStrategy.sol:128]: @brkomir I don't get this. The funds have to be withdrawn from the Yearn Vault

Comment (dxganta) [reply contracts/strategy/yearn/YearnStrategy.sol:128]: actually the earlier code, while withdrawing was looking for if our strategy contract had any uninvested funds first then withdrawing from the yearn vault only the amount minus whatever uninvested lusd we had in the yearn strategy contract. But the uninvested lusd in our yearn strategy contract will always be zero since the vault contract automatically invests everything after sending lusd to our strategy contract. So he removed that part where we are checking for uninvested funds and directly withdraws whatever required directly from yearn vault.

Comment (brkomir) [reply contracts/strategy/yearn/YearnStrategy.sol:128]: @gabrielpoca yep, as @dxganta said it, is not possible for us to end up with uninvested funds in the yearn strategy contract

Comment (gabrielpoca) [reply contracts/strategy/yearn/YearnStrategy.sol:128]: I see 👍

**Review by gabrielpoca — APPROVED** [review]
